arroyo/processing/strategies/reduce.py

```diff
@@ -115,6 +115,10 @@ def submit(self, message: Message[TPayload]) -> None:
         """
         assert not self.__closed
 
+        if isinstance(message.payload, FilteredPayload):
```

Comment:
I discovered this is broken -- it forwards messages to the next step out of order, which causes further problems down the road